Web3 don't work with latest of React

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I have an issue with web3 with latest version of react. 
It can't be imported in code.

The message is: 

```
Compiled with problems:X

ERROR in ./node_modules/browserify-sign/node_modules/safe-buffer/index.js 4:13-30

Module not found: Error: Can't resolve 'buffer' in 'project/node_modules/browserify-sign/node_modules/safe-buffer'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "buffer": require.resolve("buffer/") }'
	- install 'buffer'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "buffer": false }

 @ ./node_modules/xhr2-cookies/dist/index.js 11:9-38
 @ ./node_modules/web3-providers-http/lib/index.js 27:11-49
 @ ./node_modules/web3-core-requestmanager/lib/index.js 56:16-46
 @ ./node_modules/web3-core/lib/index.js 23:23-58
 @ ./node_modules/web3/lib/index.js 32:11-31
 @ ./src/store/user-context.js 7:0-24
 @ ./src/index.js 9:0-59 13:38-57

25 errors have detailed information that is not shown.
Use 'stats.errorDetails: true' resp. '--stats-error-details' to show it.

webpack 5.65.0 compiled with 25 errors and 27 warnings in 261 ms

```

### Expected Behavior

I am expected that the import with 

`import Web3 from 'web3';`

still work

### Steps to Reproduce

1) Create a new React Project: npx create-react-app newsite
2) enter in project: cd newSite
3) Install web3: npm i web3
4) Import web3 in App.js: import Web3 from 'web3';

### Web3.js Version

1.6.0

### Environment

- Operating System:  MacOS Monterey Macbook Pro 16" M1 Pro
- Browser: Brave
- Node.js Version: v16.13.0
- NPM Version: 8.1.0

### Anything Else?

_No response_

same here, not working and getting the following:

```
Compiled with problems:X

ERROR in ./node_modules/cipher-base/index.js 3:16-43

Module not found: Error: Can't resolve 'stream' in '/home/basti/Desktop/activate/kolor-ui/node_modules/cipher-base'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
	- install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "stream": false }


ERROR in ./node_modules/eth-lib/lib/bytes.js 7:193-227

Module not found: Error: Can't resolve 'crypto' in '/home/basti/Desktop/activate/kolor-ui/node_modules/eth-lib/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "crypto": require.resolve("crypto-browserify") }'
	- install 'crypto-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "crypto": false }


ERROR in ./node_modules/ethereumjs-util/dist.browser/account.js 71:31-48

Module not found: Error: Can't resolve 'assert' in '/home/basti/Desktop/activate/kolor-ui/node_modules/ethereumjs-util/dist.browser'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "assert": require.resolve("assert/") }'
	- install 'assert'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "assert": false }


ERROR in ./node_modules/ethereumjs-util/dist.browser/address.js 14:31-48

Module not found: Error: Can't resolve 'assert' in '/home/basti/Desktop/activate/kolor-ui/node_modules/ethereumjs-util/dist.browser'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "assert": require.resolve("assert/") }'
	- install 'assert'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "assert": false }


ERROR in ./node_modules/ethereumjs-util/dist.browser/object.js 46:31-48

Module not found: Error: Can't resolve 'assert' in '/home/basti/Desktop/activate/kolor-ui/node_modules/ethereumjs-util/dist.browser'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "assert": require.resolve("assert/") }'
	- install 'assert'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "assert": false }


ERROR in ./node_modules/swarm-js/node_modules/eth-lib/lib/bytes.js 9:193-227

Module not found: Error: Can't resolve 'crypto' in '/home/basti/Desktop/activate/kolor-ui/node_modules/swarm-js/node_modules/eth-lib/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "crypto": require.resolve("crypto-browserify") }'
	- install 'crypto-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "crypto": false }


ERROR in ./node_modules/web3-eth-accounts/lib/index.js 31:74-91

Module not found: Error: Can't resolve 'crypto' in '/home/basti/Desktop/activate/kolor-ui/node_modules/web3-eth-accounts/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "crypto": require.resolve("crypto-browserify") }'
	- install 'crypto-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "crypto": false }


ERROR in ./node_modules/web3-providers-http/lib/index.js 30:11-26

Module not found: Error: Can't resolve 'http' in '/home/basti/Desktop/activate/kolor-ui/node_modules/web3-providers-http/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "http": require.resolve("stream-http") }'
	- install 'stream-http'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "http": false }


ERROR in ./node_modules/web3-providers-http/lib/index.js 32:12-28

Module not found: Error: Can't resolve 'https' in '/home/basti/Desktop/activate/kolor-ui/node_modules/web3-providers-http/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "https": require.resolve("https-browserify") }'
	- install 'https-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "https": false }


ERROR in ./node_modules/web3-providers-ws/lib/helpers.js 11:12-26

Module not found: Error: Can't resolve 'url' in '/home/basti/Desktop/activate/kolor-ui/node_modules/web3-providers-ws/lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "url": require.resolve("url/") }'
	- install 'url'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "url": false }


ERROR in ./node_modules/xhr2-cookies/dist/xml-http-request.js 37:11-26

Module not found: Error: Can't resolve 'http' in '/home/basti/Desktop/activate/kolor-ui/node_modules/xhr2-cookies/dist'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "http": require.resolve("stream-http") }'
	- install 'stream-http'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "http": false }


ERROR in ./node_modules/xhr2-cookies/dist/xml-http-request.js 39:12-28

Module not found: Error: Can't resolve 'https' in '/home/basti/Desktop/activate/kolor-ui/node_modules/xhr2-cookies/dist'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "https": require.resolve("https-browserify") }'
	- install 'https-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "https": false }


ERROR in ./node_modules/xhr2-cookies/dist/xml-http-request.js 41:9-22

Module not found: Error: Can't resolve 'os' in '/home/basti/Desktop/activate/kolor-ui/node_modules/xhr2-cookies/dist'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "os": require.resolve("os-browserify/browser") }'
	- install 'os-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "os": false }


ERROR in ./node_modules/xhr2-cookies/dist/xml-http-request.js 43:10-24

Module not found: Error: Can't resolve 'url' in '/home/basti/Desktop/activate/kolor-ui/node_modules/xhr2-cookies/dist'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "url": require.resolve("url/") }'
	- install 'url'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "url": false }
```

In js in polyfill i need to remove the "as any" directive, but now i have this error: 
```
Compiled with problems:X

ERROR in ./src/polyfill.js 3:0-32

Module not found: Error: Can't resolve 'buffer' in '/src'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "buffer": require.resolve("buffer/") }'
	- install 'buffer'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "buffer": false }


ERROR in ./node_modules/browserify-sign/node_modules/safe-buffer/index.js 4:13-30
```

I will try to add it

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -497,3 +497,4 @@ Released with 1.0.0-beta.37 code base.
 
 ### Fixed
 -  Fix a typo in the documentation for `methods.myMethod.send` (#4599)
+-  Updated README to include Webpack 5 create-react-app support instructions (#4173)

--- a/README.md
+++ b/README.md
@@ -105,26 +105,28 @@ If you are using the types in a `commonjs` module, like in a Node app, you just 
 
 ### Web3 and Create-react-app
 
-If you are using create-react-app version >=5 you may run into issues building. This is because polyfills are not included in the latest version of create-react-app.
+If you are using create-react-app version >=5 you may run into issues building. This is because NodeJS polyfills are not included in the latest version of create-react-app.
 
 ### Solution
 
 
-- Install react-app-rewired
+- Install react-app-rewired and the missing modules
 
 If you are using yarn:
 ```bash
-yarn add --dev react-app-rewired
+yarn add --dev react-app-rewired crypto-browserify stream-browserify assert stream-http https-browserify os-browserify url buffer
 ```
 
 If you are using npm:
 ```bash
-npm install --save-dev react-app-rewired
+npm install --save-dev react-app-rewired crypto-browserify stream-browserify assert stream-http https-browserify os-browserify url buffer process
 ```
 
 - Create `config-overrides.js` in the root of your project folder with the content:
 
 ```javascript
+const webpack = require('webpack');
+
 module.exports = function override(config) {
     const fallback = config.resolve.fallback || {};
     Object.assign(fallback, {
@@ -137,44 +139,17 @@ module.exports = function override(config) {
         "url": require.resolve("url")
     })
     config.resolve.fallback = fallback;
-
+    config.plugins = (config.plugins || []).concat([
+        new webpack.ProvidePlugin({
+            process: 'process/browser',
+            Buffer: ['buffer', 'Buffer']
+        })
+    ])
     return config;
 }
 ```
 
-
-- Install the missing modules
-
-If you are using yarn:
-```bash
-yarn add --dev crypto-browserify stream-browserify assert stream-http https-browserify os-browserify url
-```
-If you are using npm:
-```bash
-npm install --save-dev crypto-browserify stream-browserify assert stream-http https-browserify os-browserify url
-```
-
-- Create the file `polyfill.js` in the src folder and in the file add:
-
-```javascript
-import { Buffer } from 'buffer';
-
-window.global = window;
-global.Buffer = Buffer;
-global.process = {
-    env: { DEBUG: undefined },
-    version: '',
-    nextTick: require('next-tick')
-};
-```
-
-- In `index.js` , import `polyfill.js` before `web3`:
-```javascript
-import './polyfill.js';
-import Web3 from 'web3';
-```
-
-- Within `package.json` change the scripts field, instead of `react-scripts` replace it with `react-app-rewired`
+- Within `package.json` change the scripts field for start, build and test. Instead of `react-scripts` replace it with `react-app-rewired`
 
 before: 
 ```typescript
@@ -182,7 +157,8 @@ before:
     "start": "react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test",
-  ...
+    "eject": "react-scripts eject"
+},
 ```
 
 after:
@@ -191,16 +167,17 @@ after:
     "start": "react-app-rewired start",
     "build": "react-app-rewired build",
     "test": "react-app-rewired test",
-  ...
+    "eject": "react-scripts eject"
+},
 ```
 
-The missing polyfills should be included now and your app should be functional with web3.
+The missing Nodejs polyfills should be included now and your app should be functional with web3.
 - If you want to hide the warnings created by the console:
 
 In `config-overrides.js` within the `override` function, add:
 
 ```javascript
-config.ignoreWarnings = [/Failed to parse source map/];`
+config.ignoreWarnings = [/Failed to parse source map/];
 ```
 
 ### Web3 and Angular

--- a/README.md
+++ b/README.md
@@ -103,6 +103,101 @@ If you are using the types in a `commonjs` module, like in a Node app, you just 
 
 ## Trouble shooting and known issues.
 
+### Web3 and Create-react-app
+
+If you are using create-react-app version >=5 you may run into issues building. This is because polyfills are not included in the latest version of create-react-app.
+
+### Solution
+
+1. Install react-app-rewired
+
+If you are using yarn:
+```
+yarn add --dev react-app-rewired
+```
+
+If you are using npm:
+```
+npm install --save-dev react-app-rewired
+```
+
+2. Create `config-overrides.js` in the root of your project folder with the content:
+
+```
+module.exports = function override(config) {
+    const fallback = config.resolve.fallback || {};
+    Object.assign(fallback, {
+        "crypto": require.resolve("crypto-browserify"),
+        "stream": require.resolve("stream-browserify"),
+        "assert": require.resolve("assert"),
+        "http": require.resolve("stream-http"),
+        "https": require.resolve("https-browserify"),
+        "os": require.resolve("os-browserify"),
+        "url": require.resolve("url")
+    })
+    config.resolve.fallback = fallback;
+
+    return config;
+}
+```
+
+3. Install the missing modules
+
+If you are using yarn:
+```
+yarn add --dev crypto-browserify stream-browserify assert stream-http https-browserify os-browserify url
+```
+If you are using npm:
+```
+npm install --save-dev crypto-browserify stream-browserify assert stream-http https-browserify os-browserify url
+```
+
+4. Create the file `polyfill.js` in the src folder and in the file add:
+
+```
+import { Buffer } from 'buffer';
+
+window.global = window;
+global.Buffer = Buffer;
+global.process = {
+    env: { DEBUG: undefined },
+    version: '',
+    nextTick: require('next-tick')
+};
+```
+
+5. In `index.js` , import `polyfill.js` before `web3`:
+```
+import './polyfill.js';
+import Web3 from 'web3';
+```
+
+6. Within `package.json` change the scripts field:
+replace 
+```
+"scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test"
+  }
+```
+with
+```
+"scripts": {
+    "start": "react-app-rewired start",
+    "build": "react-app-rewired build",
+    "test": "react-app-rewired test"
+  }
+```
+The missing polyfills should be included now and your app should be functional with web3.
+
+If you want to hide the warnings created by the console:
+
+In `config-overrides.js` add:
+
+`config.ignoreWarnings = [/Failed to parse source map/];`
+
+
 ### Web3 and Angular
 
 ### New solution

--- a/README.md
+++ b/README.md
@@ -109,21 +109,22 @@ If you are using create-react-app version >=5 you may run into issues building. 
 
 ### Solution
 
-1. Install react-app-rewired
+
+- Install react-app-rewired
 
 If you are using yarn:
-```
+```bash
 yarn add --dev react-app-rewired
 ```
 
 If you are using npm:
-```
+```bash
 npm install --save-dev react-app-rewired
 ```
 
-2. Create `config-overrides.js` in the root of your project folder with the content:
+- Create `config-overrides.js` in the root of your project folder with the content:
 
-```
+```javascript
 module.exports = function override(config) {
     const fallback = config.resolve.fallback || {};
     Object.assign(fallback, {
@@ -141,20 +142,21 @@ module.exports = function override(config) {
 }
 ```
 
-3. Install the missing modules
+
+- Install the missing modules
 
 If you are using yarn:
-```
+```bash
 yarn add --dev crypto-browserify stream-browserify assert stream-http https-browserify os-browserify url
 ```
 If you are using npm:
-```
+```bash
 npm install --save-dev crypto-browserify stream-browserify assert stream-http https-browserify os-browserify url
 ```
 
-4. Create the file `polyfill.js` in the src folder and in the file add:
+- Create the file `polyfill.js` in the src folder and in the file add:
 
-```
+```javascript
 import { Buffer } from 'buffer';
 
 window.global = window;
@@ -166,37 +168,40 @@ global.process = {
 };
 ```
 
-5. In `index.js` , import `polyfill.js` before `web3`:
-```
+- In `index.js` , import `polyfill.js` before `web3`:
+```javascript
 import './polyfill.js';
 import Web3 from 'web3';
 ```
 
-6. Within `package.json` change the scripts field:
-replace 
-```
+- Within `package.json` change the scripts field, instead of `react-scripts` replace it with `react-app-rewired`
+
+before: 
+```typescript
 "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
-    "test": "react-scripts test"
-  }
+    "test": "react-scripts test",
+  ...
 ```
-with
-```
+
+after:
+```typescript
 "scripts": {
     "start": "react-app-rewired start",
     "build": "react-app-rewired build",
-    "test": "react-app-rewired test"
-  }
+    "test": "react-app-rewired test",
+  ...
 ```
+
 The missing polyfills should be included now and your app should be functional with web3.
+- If you want to hide the warnings created by the console:
 
-If you want to hide the warnings created by the console:
+In `config-overrides.js` within the `override` function, add:
 
-In `config-overrides.js` add:
-
-`config.ignoreWarnings = [/Failed to parse source map/];`
-
+```javascript
+config.ignoreWarnings = [/Failed to parse source map/];`
+```
 
 ### Web3 and Angular
 

